core/src/main/java/com/google/cloud/sql/core/Connector.java

```diff
@@ -154,8 +156,21 @@ DefaultConnectionInfoCache getConnection(ConnectionConfig config) {
   private DefaultConnectionInfoCache createConnectionInfo(ConnectionConfig config) {
     logger.debug(
         String.format("[%s] Connection info added to cache.", config.getCloudSqlInstance()));
-    return new DefaultConnectionInfoCache(
-        config, adminApi, instanceCredentialFactory, executor, localKeyPair, minRefreshDelayMs);
+    if (config.getConnectorConfig().getRefreshStrategy() == RefreshStrategy.LAZY) {
+      // Resolve the key operation immediately.
+      KeyPair keyPair = null;
+      try {
+        keyPair = localKeyPair.get();
+      } catch (InterruptedException | ExecutionException e) {
+        throw new RuntimeException(e);
+      }
+      return new DefaultConnectionInfoCache(
```

Comment:
Looking more closely at this, I see we're missing a ConnectionInfoCache interface in this repo (which exists in Go and in AlloyDB Java). Instead of having multiple constructors for one cache type, we should be introducing separate caches that both implement a ConnectionInfoCache interface. This makes for better separation of concerns. The refresher strategy might still be useful, but the most important thing it to think of connection info cache as being the interface for which we have multiple implementations.